### PR TITLE
Ptero process view

### PR DESCRIPTION
**Pull request by davidlmorton**
This allows Processes run by PTero to be viewed in the same way as any other Process ('genome process view').  Below are a few examples:

```
$ genome process view ebd5d6d2f1ff4a3eac83feb628dc655d
Using libraries at /home/archive/dmorton/genome/lib/perl
'process' may require verification...
Resolving parameter 'process' from command argument 'ebd5d6d2f1ff4a3eac83feb628dc655d'... found 1
=== Process ===
Process ID: ebd5d6d2f1ff4a3eac83feb628dc655d    Process Status: Succeeded
Process Started: 2015-06-08 17:07:13            Process Ended: 2015-06-08 17:57:36
Run by: apipe-tester

Process Class: Genome::VariantReporting::Process::CreateReport
Software Revision: /gscuser/dmorton/deployments/default/genome/lib/perl/Genome/Site/TGI/SiteLib:/gscuser/dmorton/deployments/default/genome/lib/perl:/gsc/scripts/opt/genome/snapshots/genome-3605/lib/perl:/etc/perl:/usr/local/lib/site_perl
SoftwareResult Test Name(s): "dmorton_ptero_test" (11), undef (2)
MetaData Directory: /gscmnt/gc9020/info/model_data/process/ebd5d6d2f1ff4a3eac83feb628dc655d

=== Workflow ===
   STAGE     STATUS              STARTED       DURATION  P-INDEX  NAME
          succeeded  2015-06-08T22:06:53       00:50:14           Genome::Process(ebd5d6d2f1ff4a3eac83feb628dc655d)
          succeeded  2015-06-08T22:06:54       00:31:31           . vep
 execute  succeeded  2015-06-08T22:07:03       00:31:22           . . Run vep
          succeeded  2015-06-08T22:38:26       00:06:06           . BamReadcount
shortcut  succeeded  2015-06-08T22:38:27       00:00:09           . . Get inputs from plan
                parallel-by: aligned_bam_result_id                . . Run bam-readcount
 execute  succeeded  2015-06-08T22:38:46       00:04:45  0        . . . Run bam-readcount
 execute  succeeded  2015-06-08T22:43:38       00:00:53           . . Annotate vcf with readcounts
          succeeded  2015-06-08T22:44:32       00:06:18           . dbsnp
 execute  succeeded  2015-06-08T22:45:14       00:05:36           . . Run dbsnp
          succeeded  2015-06-08T22:50:51       00:02:07           . homopolymer
 execute  succeeded  2015-06-08T22:50:59       00:01:58           . . Run homopolymer
          succeeded  2015-06-08T22:52:58       00:00:41           . nhlbi
 execute  succeeded  2015-06-08T22:53:07       00:00:31           . . Run nhlbi
 execute  succeeded  2015-06-08T22:54:00       00:00:50           . Generate Report (acmg)
 execute  succeeded  2015-06-08T22:54:02       00:00:40           . Generate Report (bed)
 execute  succeeded  2015-06-08T22:53:59       00:03:07           . Generate Report (full)
 execute  succeeded  2015-06-08T22:53:59       00:00:45           . Generate Report (simple)
 execute  succeeded  2015-06-08T22:53:57       00:01:17           . Generate Report (vcf)
```

Here, there was an error: the process was run as a general user, and the shell-command workers were running as apipe-tester (instead of root).
```
$ genome process view 5481d3b791e74ff4a5d8d2d5386447e1
Using libraries at /home/archive/dmorton/genome/lib/perl
'process' may require verification...
Resolving parameter 'process' from command argument '5481d3b791e74ff4a5d8d2d5386447e1'... found 1
=== Process ===
Process ID: 5481d3b791e74ff4a5d8d2d5386447e1    Process Status: Scheduled
Process Started:                                Process Ended:
Run by: dmorton

Process Class: Genome::VariantReporting::Process::CreateReport
Software Revision: /gscuser/dmorton/deployments/default/genome/lib/perl/Genome/Site/TGI/SiteLib:/gscuser/dmorton/deployments/default/genome/lib/perl:/gsc/scripts/opt/genome/snapshots/genome-3605/lib/perl:/etc/perl:/usr/local/lib/site_perl
SoftwareResult Test Name(s): No results found
MetaData Directory: /gscmnt/gc9021/info/model_data/process/5481d3b791e74ff4a5d8d2d5386447e1

=== Workflow ===
   STAGE     STATUS              STARTED       DURATION  P-INDEX  NAME
             failed  2015-06-05T19:07:23       00:00:03           Genome::Process(5481d3b791e74ff4a5d8d2d5386447e1)
             failed  2015-06-05T19:07:24       00:00:02           . vep
shortcut    errored  2015-06-05T19:07:25       00:00:00           . . Run vep
                                                                  . BamReadcount
                                                                  . . Get inputs from plan
                parallel-by: aligned_bam_result_id                . . Run bam-readcount
                                                                  . . Annotate vcf with readcounts
                                                                  . dbsnp
                                                                  . . Run dbsnp
                                                                  . homopolymer
                                                                  . . Run homopolymer
                                                                  . nhlbi
                                                                  . . Run nhlbi
                                                                  . Generate Report (acmg)
                                                                  . Generate Report (bed)
                                                                  . Generate Report (full)
                                                                  . Generate Report (simple)
                                                                  . Generate Report (vcf)

Name: Run vep.shortcut.44001    Status: errored
Error Message: Failed to setreuid: Operation not permitted
Stdout:
Stderr:
```

This example shows log files are displayed for 'running'/'failed' steps.  The time formatting issue is a little bug related to the database being in UTC and local time being used to calculate the duration of a step.
```
$ genome process view fd1a5739390b41fdacde5a24e1358cd2
Using libraries at /home/archive/dmorton/genome/lib/perl
'process' may require verification...
Resolving parameter 'process' from command argument 'fd1a5739390b41fdacde5a24e1358cd2'... found 1
=== Process ===
Process ID: fd1a5739390b41fdacde5a24e1358cd2    Process Status: Running
Process Started: 2015-06-09 09:30:46            Process Ended:
Run by: apipe-tester

Process Class: Genome::VariantReporting::Process::CreateReport
Software Revision: /gscuser/dmorton/deployments/default/genome/lib/perl/Genome/Site/TGI/SiteLib:/gscuser/dmorton/deployments/default/genome/lib/perl:/gsc/scripts/opt/genome/snapshots/genome-3605/lib/perl:/etc/perl:/usr/local/lib/site_perl
SoftwareResult Test Name(s): "dmorton_ptero_test" (1)
MetaData Directory: /gscmnt/gc13030/info/model_data/process/fd1a5739390b41fdacde5a24e1358cd2

=== Workflow ===
   STAGE     STATUS              STARTED       DURATION  P-INDEX  NAME
            running  2015-06-09T14:30:26       -4:-58:-35           Genome::Process(fd1a5739390b41fdacde5a24e1358cd2)
          succeeded  2015-06-09T14:30:26       00:00:33           . vep
shortcut  succeeded  2015-06-09T14:30:27       00:00:31           . . Run vep
            running  2015-06-09T14:31:00       -4:-59:-9           . BamReadcount
shortcut  succeeded  2015-06-09T14:31:01       00:00:18           . . Get inputs from plan
                parallel-by: aligned_bam_result_id                . . Run bam-readcount
shortcut    running  2015-06-09T14:31:20       -4:-59:-29  0        . . . Run bam-readcount
                                                                  . . Annotate vcf with readcounts
                                                                  . dbsnp
                                                                  . . Run dbsnp
                                                                  . nhlbi
                                                                  . . Run nhlbi
                                                                  . Generate Report (acmg)
                                                                  . Generate Report (bed)
                                                                  . Generate Report (full)
                                                                  . Generate Report (simple)
                                                                  . Generate Report (vcf)

Name: Run bam-readcount.shortcut.44122    Status: running
Stdout Log: /gscmnt/gc13030/info/model_data/process/fd1a5739390b41fdacde5a24e1358cd2/logs/Run_bam-readcount.shortcut.44122.out
Stderr Log: /gscmnt/gc13030/info/model_data/process/fd1a5739390b41fdacde5a24e1358cd2/logs/Run_bam-readcount.shortcut.44122.err
```

### CODE REVIEW

File: lib/perl/Genome/WorkflowBuilder/DAG.pm [hunk 1]
```diff
@@ -210,9 +211,10 @@ sub get_ptero_builder_task {
 sub get_ptero_builder_for_process {
     require Ptero::Builder::Workflow;
 
-    my ($self, $process_id) = Params::Validate::validate_pos(@_, 1, 1);
+    my ($self, $process) = Params::Validate::validate_pos(@_, 1, 1);
+    my $process_id = $process->id;
```
Comment (mkiwala) [inline lib/perl/Genome/WorkflowBuilder/DAG.pm]: It does not appear that this `$process_id` is used any longer, so line 215 could be deleted.

Comment (davidlmorton) [inline lib/perl/Genome/WorkflowBuilder/DAG.pm]: They are just below the cut that github puts in:

lines 221 223 228 and 269

**Comment by davidlmorton** [comment]
Thats a good catch on the 'unstarted' for failed steps, I missed it, and only caught it in the wild when a failed step didn't show up as expected... fix incoming.

**Comment by mkiwala** [comment]
+1

**Pull request merged**